### PR TITLE
Fix rubocop / rspec

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
Fix some sigv4 spec failures + rubocop error.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gems/aws-sigv4/spec/aws-sigv4/signer_spec.rb [hunk 1]
```diff
@@ -5,6 +5,24 @@
 require 'tempfile'
 require 'base64'
```
Comment (mullermp) [inline gems/aws-sigv4/spec/aws-sigv4/signer_spec.rb:7]: Isn't this defined in spec helper already? Also I think you just need to use defined?

Comment (alextwoods) [reply gems/aws-sigv4/spec/aws-sigv4/signer_spec.rb]: It was but wasn't working correctly, I've fixed it.

File: gems/aws-sdk-core/spec/spec_helper.rb [hunk 2]
```diff
@@ -33,6 +33,13 @@ def initialize(_config = Config.new, _options = {})
   Config = Struct.new(:region, keyword_init: true)
 end
 
+module Aws
```
Comment (mullermp) [inline gems/aws-sdk-core/spec/spec_helper.rb:36]: Do you need defined? check here as well?

**Review by mullermp — APPROVED** [review]

File: gems/aws-sdk-core/spec/spec_helper.rb (hunk 2 above)
Comment (alextwoods) [reply gems/aws-sdk-core/spec/spec_helper.rb:36]: There is no consequence for redefining these - however, it becomes a tricky order issue to check defined? We would need to add a `unless defined?(Aws::Crt::Checksums) && defined?(Aws::Crt::NextFeature) && defined?(Aws::Crt::Feature3)` for everything we add here...

**Pull request merged**